### DISCUSSION TITLE
🤔 Typing named slots & autocompleltion?

### DISCUSSION THREAD

**Discussion by jd-solanki** [vue-macros/vue-macros#265] [Q&A]
Hi,

For Anu, I want to type slots and want to extract slot object for other components. Thanks to you, we have `defineSlots` but how to define a slot that isn't scoped slot?

Thanks.

**Comment by jd-solanki** [comment]
Will this work?
```ts
defineSlots<{
  title: undefined
}>()
```

If I check component meta I do get title but I don't get autocompletion for slot names and don't get a warning/error for using slots that aren't defined.

![image](https://user-images.githubusercontent.com/47495003/218262512-0dac62aa-eb0a-4ce6-bbd5-3ad2eb4ab4ad.png)

**Reply by sxzz** [reply]
Just make it an empty object
https://github.com/elk-zone/elk/blob/328fd1bf7094df57eade0f2a90516bcf722572a7/components/nav/NavSideItem.vue#L12-L15

**Reply by jd-solanki** [reply]
Still, I don't get warning/error for an undefined slot:

![image](https://user-images.githubusercontent.com/47495003/218293557-071c0f08-46fc-4f66-a26b-818017b8b430.png)

my slots:
```ts
defineSlots<{
  'label': {}
  'prepend': {}
  'prepend-inner': {}
  'append-inner': {}
  'append': {}
  'bottom': {}
}>()
```

Do I have to do any additional setup?

```json
 "compilerOptions: {
    "types": [
      "vitest/globals",
      "unplugin-vue-macros/macros-global"
    ]
  },
  "vueCompilerOptions": {
    "plugins": [
      "@vue-macros/volar/define-model",
      "@vue-macros/volar/define-props",
      "@vue-macros/volar/define-props-refs",
      "@vue-macros/volar/short-vmodel",
      "@vue-macros/volar/define-slots",
      "@vue-macros/volar/export-props"
    ]
  },
```

**Comment by jd-solanki** [comment]
I also tried this in a new project and still not getting autocompletion or errors.

Do you want a reproduction, I just followed installation steps in new project creating using `pnpm create vue@latest`?

**Comment by jd-solanki** [comment]
@sxzz I guess it's volar issue, is it?

**Reply by sxzz** [reply]
Could you please provide a minimum repo?

**Reply by jd-solanki** [reply]
Yeah sure, Here it is: https://github.com/jd-solanki/defineSlots-volar-support

There are two defined slots and one undefined slot in `ABtn` component.

Thanks for your interest 😇

**Reply by jd-solanki** [reply]
I guess we should better track in issues as this is not a discussion/question anymore.

Thanks for your respond 😇
